### DISCUSSION TITLE
Can a component be notified when a configured theme screen breakpoint is matched?

### DISCUSSION THREAD

**Discussion by jgornick** [nativewind/nativewind#298] [Q&A] [answered]
We want our components to use the same implementation for when NativeWind applies a responsive class to a component, but programmatically. 

As an example: 

```typescript
// this component will re-render when NativeWind has detected the screen size changed to a configured theme screen size.
const { screen } = useNativeWindScreens()

console.log(screen) // 'lg'
```

Is this exposed someway via NativeWind or NativeWind 3.x?

**Comment by jgornick** [comment]
This idea could also apply to other media related queries like orientation.

There is already the `useColorScheme` hook that is similar to what we're looking to do.

**Comment by marklawlor** [comment, marked as answer]
I'm going to add this to the Requested Features project, but I'll give some context to why it doesn't currently exist.

`withLayout` would require the runtime JavaScript to know about the contents of the `tailwind.config.js`. For Native this is fine - NativeWind processes the styles so it could add extra meta data for the breakpoints.

However, for web, the processing is done by an external tool and NativeWind works as a compatibility layer. A hacky way for this to work would be for NativeWind to use the Tailwind preset to inject the breakpoint into information into a CSS variable which could be read at runtime.

"Can't the hook just import the config file" - the config file can move and might require Node.js APIs like `fs`, it's just not that simple.

For this reason it's probably better for users to create their own hook, as it's easier for them to control the complexities of reading `tailwind.config.js`.

A possible "solution" could be a how-to guide on the website.

`useColorTheme` exists because manually setting the colour theme exists on web but not on native. NativeWind implements it's own version of the RN `Appearance` module and needed a way for users to read/write it's value.

`useOrientation` is interesting, but this functionality is already provided by React Native and polyfilled in React Native Web. I think I might just write a guide on it as well.

There is another aspect to this as well, where does the scope for NativeWind end? I want to keep the scope rather contained and just be about processing `className` but things like `useColorScheme` and `styled()` challenge that.

**Reply by jgornick** [reply]
Makes a lot of sense @marklawlor! I totally understand the scope challenges that **could** happen if NativeWind started to expose more of it's internals. 

We will look at creating something cross-platform on our side and share the approach we took.

Thanks again!

**Reply by andrioid** [reply]
@jgornick What did you end up with?

**Reply by jgornick** [reply]
We ended up pivoting and simply point our consumers to utilize the variant classes.

**Comment by danstepanov** [comment]
Closing this discussion, if this is still relevant, feel free to open a new one.
